XWOpt/Vector2Adapter.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * Copyright 2017 Jason McNew
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies
+ * or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+using System;
+using System.Collections.ObjectModel;
+using System.Reflection;
+
+namespace SchmooTech.XWOpt
+{
+    class Vector2Adapter<TVector2> : VectorAdapter
+    {
+        // TODO: Call constructor if ref type.
+        ConstructorInfo vector3Cotr;
+        FieldInfo X, Y;
+
+        public Vector2Adapter()
+        {
+            vector3Cotr = typeof(TVector2).GetConstructor(new Type[] { typeof(float), typeof(float) });
+
+            X = typeof(TVector2).GetField("X", BindingFlags.Instance | BindingFlags.IgnoreCase | BindingFlags.Public);
+            Y = typeof(TVector2).GetField("Y", BindingFlags.Instance | BindingFlags.IgnoreCase | BindingFlags.Public);
+
+            if (null == X || null == Y)
+            {
+                throw new ArgumentException("Vector2 type must have x, y, and z fields (case insensitive).");
```

Comment:
.... where's the Z check?

Reply:
Bad copy/pasta from Vector3Adapter. Fix the error message.

Reply:
Fixed, thanks!